### PR TITLE
Change permissions of stdlib pkgimgs

### PR DESCRIPTION
**Pull request by gbaraldi**

### CODE REVIEW

**Review by vchuravy — APPROVED** [review]

**Comment by gbaraldi** [comment]
Maybe? 
Elliot said:
> It is understandable that the pkgimage generation code only assigns the user executable bit, since it is, in general, sitting in a user’s home directory. However, since we are shipping a distribution here, we should alter the executable bits at make install time.
which seems reasonable.

**Comment by staticfloat** [comment]
I think if you’re okay with making it more permissive, that would avoid one potential stumbling block in case sys admins want to run a shared depot.

**Comment by vchuravy** [comment]
Yeah, I see no issue with that. The place to change it is https://github.com/JuliaLang/julia/blob/82fbf54131fae6dbb74c0a125c68eecf46285154/base/loading.jl#L2087

File: base/loading.jl [hunk 1]
```diff
@@ -2084,7 +2084,7 @@ function compilecache(pkg::PkgId, path::String, internal_stderr::IO = stderr, in
                 # Ensure that the user can execute the `.so` we're generating
                 # Note that on windows, `filemode(path)` typically returns `0o666`, so this
                 # addition of the execute bit for the user is doubly needed.
-                chmod(tmppath_so, filemode(path) & 0o777 | 0o300)
+                chmod(tmppath_so, filemode(path) & 0o777 | 0o755)
```
Comment (staticfloat) [inline base/loading.jl:2087]:
```suggestion
                chmod(tmppath_so, filemode(path) & 0o777 | 0o333)
```

Let's keep the comment here truthful; we're only adding the executable bits.

**Comment by staticfloat** [comment]
> `julia-58d9c87fc5/share/julia/compiled/v1.10/LLD_jll/ZHBMJ_epoh3.dylib: replacing existing signature`

Codesigning changes are [looking good!](https://buildkite.com/julialang/julia-master/builds/19573#018559de-45d1-4b28-8b7d-6866eb1ab3da/474-530)

> `The staple and validate action worked!`

[Notarization worked](https://buildkite.com/julialang/julia-master/builds/19573#018559de-45d1-4b28-8b7d-6866eb1ab3da/564-634), thanks Gabriel!

**Pull request merged**